oolab/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -1,9 +1,27 @@
 package agh.ics.oop;
 
+import java.util.Objects;
+
+import static agh.ics.oop.MapDirection.*;
+
 public class Animal {
 
-    private MapDirection orientation = MapDirection.NORTH;
-    private Vector2d position = new Vector2d(2,2);
+    private MapDirection orientation;
+    private Vector2d position;
+    static IWorldMap map;
```

Comment:
dlaczego `static`? wiem, że czasem kusi "bo przecież taki obiekt będzie jeden w aplikacji", ale z perspektywy zwierzątka to ono nie wie ile tam gdzieś na świecie jest map, ono dostaje mapę, którą ma sobie przytulić i nie czynić żadnych dodatkowych założeń. Generalnie statyczne atrybuty (inne niż stałe) to największe zło i polecam tego unikać, z podobnych powodów jak np. zmiennych globalnych w C/C++.